Userland/Applications/Browser/Tab.cpp

```diff
@@ -446,6 +449,33 @@ Tab::Tab(Type type)
     add_search_engine("Google", "https://google.com/search?q={}");
     add_search_engine("Yandex", "https://yandex.com/search/?text={}");
 
+    auto custom_search_engine = GUI::Action::create_checkable("Custom", [&](auto& action) {
+        String search_engine;
+        if (GUI::InputBox::show(window(), search_engine, "Enter Format:", "Custom Search Engine") != GUI::InputBox::ExecOK || search_engine.is_empty()) {
+            m_disable_search_engine->activate();
+            return;
+        }
+
+        int argument_count = search_engine.replace("{}", "{}", true);
```

Comment:
What does this mean?

Reply:
This is the way to count occurences of substring in a string - `String::replace()` returns count of replaces. I didn't find another one-line way to do this.

Reply:
doing a bunch of useless memory writes doesnt sound like a great idea, how about adding a new String::count method?

Reply:
It's a good idea I think :^)

Reply:
I added two new methods, `String::count` and `String::find_all`. `String::replace` uses now `String::find_all`.